sensors/simsensor_common/event_queue.c

```diff
@@ -0,0 +1,98 @@
+/*
+ * Phoenix-RTOS
+ *
+ * Queue for buffering sensor event data from reader to simsensors
+ *
+ * Copyright 2023 Phoenix Systems
+ * Author: Piotr Nieciecki
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include "event_queue.h"
+
+#include <stdlib.h>
+
+
+int eventQueue_init(event_queue_t *q, unsigned int cnt)
+{
+	if (cnt == 0 || q == NULL) {
+		return -1;
+	}
+
+	q->queue = malloc(sizeof(sensor_event_t) * cnt);
```

Comment:
If we can live with the cost of per object malloc imho list is a better choice for this

Reply:
I will stick to this implementation. For our use case queue with constant size is enough. Moreover we are concerned about performance, some sensors can report new measurements really often.

Reply:
Ack